packages/core/src/routes/interaction/additional.ts

```diff
@@ -204,4 +211,44 @@ export default function additionalRoutes<T extends IRouterParamContext>(
       throw new Error('Not implemented');
     }
   );
+
+  router.post(
+    `${interactionPrefix}/${verificationPath}/webauthn-authentication`,
+    koaGuard({
+      status: [200],
+      response: webAuthnAuthenticationOptionsGuard,
+    }),
+    async (ctx, next) => {
+      const { interactionDetails, createLog } = ctx;
+      // Check interaction exists
+      const interaction = getInteractionStorage(interactionDetails.result);
+      const { event, accountId } = interaction;
+      assertThat(
+        event === InteractionEvent.SignIn && accountId,
+        new RequestError({
+          code: 'session.mfa.mfa_sign_in_only',
+        })
+      );
+      createLog(`Interaction.${event}.Mfa.WebAuthn.Create`);
```

Comment:
should we move the log creation before the assertion? or else we can not record the failure in our log?

Reply:
No, we have to narrow down the type, because there is no "forgot password" log type.